versiondb/tsrocksdb/store.go

```diff
@@ -26,10 +26,15 @@ var (
 
 	_ versiondb.VersionStore = Store{}
 
-	defaultWriteOpts = grocksdb.NewDefaultWriteOptions()
-	defaultReadOpts  = grocksdb.NewDefaultReadOptions()
+	defaultWriteOpts     = grocksdb.NewDefaultWriteOptions()
+	defaultSyncWriteOpts = grocksdb.NewDefaultWriteOptions()
+	defaultReadOpts      = grocksdb.NewDefaultReadOptions()
 )
 
+func init() {
+	defaultSyncWriteOpts.SetSync(true)
```

Comment:
will we wait the sync job get done before app get restart?

Reply:
Sync will finish before the Write call returns. But because versiondb is written after iavl, so if it's interrupted between, versiondb will lag behind one version, but in that rare case user can use normal rollback command to rollback iavl by one version.
There's another case for mismatch, memiavl async commit, in that case memiavl can lag behind versiondb, I think it's safe to allow this case to happen, so we should simply relax the version check on start up to `require(versiondb version >= iavl version)`

Reply:
yes, but the case like [disable](https://github.com/crypto-org-chain/cronos/pull/1300/files#diff-ac33edb186128c1aa7c7d1d4ceecce554b6c4e5c2c22d065d80e64b65c2b3b6eR121) and enable version db, user have to skip or recover in other ways

Reply:
> yes, but the case like [disable](https://github.com/crypto-org-chain/cronos/pull/1300/files#diff-ac33edb186128c1aa7c7d1d4ceecce554b6c4e5c2c22d065d80e64b65c2b3b6eR121) and enable version db, user have to skip or recover in other ways

they have to manually recover, if not lag behind too much, the easiest way is still rollback iavl and sync.